### PR TITLE
Added a note about how to enable http_method_override for caching kernels

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.3+ |
| Fixed tickets | #3569 |

### CODE REVIEW

File: book/http_cache.rst [hunk 1]
```diff
@@ -163,6 +163,23 @@ kernel::
 The caching kernel will immediately act as a reverse proxy - caching responses
 from your application and returning them to the client.
 
+..caution::
```
Comment (WouterJ) [inline book/http_cache.rst]: missing space between `..` and `caution`

File: book/http_cache.rst [hunk 2]
```diff
@@ -163,6 +163,23 @@ kernel::
...
+
+    By default, the caching kernel ignores the ``framework.http_method_override``
```
Comment (WouterJ) [inline book/http_cache.rst]: caching kernel -> cached kernel

File: book/http_cache.rst [hunk 3]
```diff
@@ -163,6 +163,23 @@ kernel::
...
+    option, which could lead to errors when using ``PUT``, ``DELETE`` and ``PURGE``
+    methods in HTTP requests.
+
+    Invoke the ``enableHttpMethodParameterOverride()`` method before creating the
+    ``Request`` object in order to take this option into account::
+
+        // web/app.php
+        // ...
```
Comment (WouterJ) [inline book/http_cache.rst]: there should be an empty line between a file comment and any other comment

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -77,9 +77,12 @@ http_method_override
 This determines whether the ``_method`` request parameter is used as the intended
 HTTP method on POST requests. If enabled, the
 :method:`Request::enableHttpMethodParameterOverride <Symfony\\Component\\HttpFoundation\\Request::enableHttpMethodParameterOverride>`
-method gets called automatically. It becomes the service container parameter
-named ``kernel.http_method_override``. For more information, see
-:doc:`/cookbook/routing/method_parameters`.
+method gets called automatically, unless the application uses a caching kernel,
+where you need to invoke that method manually (see :ref:`symfony2-reverse-proxy`
+for details).
```
Comment (WouterJ) [inline reference/configuration/framework.rst]: to make my life easier, what do you think about removing this from this PR and putting it into my config reference rewrite PR?

File: book/http_cache.rst (hunk 2 above)
Comment (javiereguiluz) [inline book/http_cache.rst]: I'm not sure about this: `cached kernel` means that the kernel itself is cached. But `caching kernel` means that is the kernel the one who caches things. I think the second definition is more accurate. But I agree that *"caching kernel"* doesn't sound great.

File: reference/configuration/framework.rst (hunk 4 above)
Comment (javiereguiluz) [inline reference/configuration/framework.rst]: I've removed this change ... but please don't forget to add it into your PR. Thanks.

File: book/http_cache.rst (hunk 2 above)
Comment (WouterJ) [inline book/http_cache.rst]: What about: "By default, a kernel based the cache ignores the ``framework.http_method_override``" ?

File: reference/configuration/framework.rst (hunk 4 above)
Comment (WouterJ) [inline reference/configuration/framework.rst]: Done: b560fab85d9f71fe00a11fd216bc9351e9f04a99

**Comment by xabbuh** [comment]
:+1:

File: book/http_cache.rst (hunk 2 above)
Comment (javiereguiluz) [inline book/http_cache.rst]: I think there is some sort of typo in your last suggestion:

> "By default, a kernel based the cache ignores the framework.http_method_override"

Comment (WouterJ) [inline book/http_cache.rst]: "a kernel based on the cache ingores [...]"

**Comment by stof** [comment]
@javiereguiluz you should remove the first 2 commits from the history of this PR by rebasing your branch

**Comment by weaverryan** [comment]
Thanks Javier! I moved some of the note into the reference section at sha: 07cb14b228fd309ec4d12cca772d3dff760eceba. If you see any issues, please let me know! Another issue closed :)

**Pull request closed without merging**